app_potato/src/serial_comm.c

```diff
@@ -0,0 +1,40 @@
+// Zephyr Includes
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#define SERIAL_TASK_STACK_SIZE 512
+
+LOG_MODULE_REGISTER(serial_comm);
+
+// Threads
+static void receive_serial(void);
+K_THREAD_DEFINE(serial_rx_thread, SERIAL_TASK_STACK_SIZE, receive_serial, NULL, NULL, NULL, K_PRIO_PREEMPT(20), 0, 1000);
+
+// Global Variables
+static uint8_t event_byte = 0;
+
+static bool init_serial() {
+
+    return true;
+}
+
+uint8_t get_event_from_serial() {
+    uint8_t ret = 0;
+
+    if (event_byte) {
+        ret = event_byte;
+        event_byte = 0;
+    }
+
+    return ret;
+}
+
+static void receive_serial(void) {
+    if (!init_serial()) {
+        return;
+    }
+
+    while (true) {
+        // TODO: Read from serial and set event_byte
+    }
```

Comment:
Empty while loop. Please resolve todo 🥺

Reply:
Dont have library support yet :)

Reply:
To clarify, we probably want a library for dealing with rs485 communications

Reply:
From some quick prototyping, this may not be as bad and may just be a raw UART call, but need more time to determine the best course of action for now. Most likely this will just uart_poll_in and see what bytes its getting. Might also just may add some more utility functions similar to the event_notification helpers for UDP.